### PR TITLE
Coverage increase: Contracts package

### LINKED ISSUE #639: Contracts

### PR DESCRIPTION
**Pull request by alshapi**
**Description**:
This PR increases the unit tests coverage of the Contracts package.

**Related issue(s)**:

Closes #639

- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/src/main/java/com/hedera/services/contracts/execution/EvmTxProcessor.java [hunk 1]
```diff
@@ -302,6 +302,7 @@ private AbstractMessageProcessor getMessageProcessor(final MessageFrame.Type typ
 			case CONTRACT_CREATION:
 				return contractCreationProcessor;
 			default:
+				//hardly testable, since the Type enum is not extendable and has fixed values
```
Comment (georgiyazovaliiski) [inline hedera-node/src/main/java/com/hedera/services/contracts/execution/EvmTxProcessor.java:305]: remove comment

File: hedera-node/src/main/java/com/hedera/services/contracts/operation/HederaSelfDestructOperation.java [hunk 2]
```diff
@@ -66,6 +66,7 @@ public OperationResult execute(final MessageFrame frame, final EVM evm) {
 			return new OperationResult(errorGasCost(null),
 					Optional.of(HederaExceptionalHaltReason.INVALID_SOLIDITY_ADDRESS));
 		} else {
+			//hardly testable
```
Comment (georgiyazovaliiski) [inline hedera-node/src/main/java/com/hedera/services/contracts/operation/HederaSelfDestructOperation.java:69]: remove comment if not going to test this one

**Review by georgiyazovaliiski — CHANGES_REQUESTED** [review]

File: hedera-node/src/main/java/com/hedera/services/contracts/operation/HederaSelfDestructOperation.java [hunk 3]
```diff
@@ -66,6 +66,7 @@ public OperationResult execute(final MessageFrame frame, final EVM evm) {
...
 			return super.execute(frame, evm);
```
[the remark this answers is not in the document]
Comment (alshapi) [reply hedera-node/src/main/java/com/hedera/services/contracts/operation/HederaSelfDestructOperation.java:70]: We reach the line, but then the real method in the superclass is executed and the test fails

**Review by joan41868 — CHANGES_REQUESTED** [review]
Let's address the changes which @georgiyazovaliiski listed.

**Pull request merged**